### PR TITLE
update

### PR DESCRIPTION
**Pull request by i-zhen**
add logout ability

### CODE REVIEW

File: src/Web/Scotty/Login/Session.hs [hunk 1]
```diff
@@ -193,6 +194,29 @@ authCheck d a = do
...
+
+-- delete cookie
+logoutSess :: (MonadIO m, ScottyError e)
+           => SessionConfig
+           -> ActionT e m ()
+logoutSess sc = do
+  vaultContents <- liftIO readVault
+  c <- SC.getCookie "SessionId"
+  case c of
+    Nothing -> status internalServerError500
```
Comment (asg0451) [inline src/Web/Scotty/Login/Session.hs:209]: this should 404, not 500.

File: src/Web/Scotty/Login/Session.hs [hunk 2]
```diff
@@ -193,6 +194,29 @@ authCheck d a = do
...
+    Just v -> do
+      let session = find (\lambda -> sessionSid lambda == T.fromStrict v) vaultContents
```
Comment (asg0451) [inline src/Web/Scotty/Login/Session.hs:211]: change 'lambda' to something useful, like 'session' or 'sesh' or even 's'

File: src/Web/Scotty/Login/Session.hs [hunk 3]
```diff
@@ -193,6 +194,29 @@ authCheck d a = do
...
+      case session of
+        Nothing -> status internalServerError500
```
Comment (asg0451) [inline src/Web/Scotty/Login/Session.hs:213]: again, 404

**Review by asg0451 — CHANGES_REQUESTED** [review]
Thanks for the PR. Some requests: please make a test case for this function, add haddock documentation (follow format in authCheck), and add this to the example haddock comment at the top of the file. other than that good shit! i never thought to add a logout route. what led you to come across my repo?

**Comment by i-zhen** [comment]
Sorry for the carelessly work. I just would like to implement the logout ability in my website. I may work for this later.

**Comment by asg0451** [comment]
no worries, i welcome future contributions. i may implement this sometime over the weekend if i get bored, i just pushed a bunch of optimizations and changes anyway. let me know if you have any other feature requests. 

out of curiosity, are you using this package? I'm using it for my site but i thought i was the only one. it's definitely not production-ready yet but i think it works well enough.

**Comment by i-zhen** [comment]
Yes, I'm using it with Scotty. I built a Hacker News-like website and it works for my site. Thanks a lot

**Pull request closed without merging**